pymatgen/analysis/diffraction/tem.py

```diff
@@ -0,0 +1,788 @@
+                specification of Debye-Waller factors. Note that these
+                factors are temperature dependent.
+            cs (float): the chromatic aberration coefficient. set by default to 1 mm.   
+            later on: may want "number of iterations", "magnification", "critical value of beam",
+            "twin direction" for certain materials, "sample thickness", and "excitation error s"
+        """
+        self.symprec = symprec
+        self.voltage = voltage
+        self.beam_direction = beam_direction
+        self.camera_length = camera_length
+        self.debye_waller_factors = debye_waller_factors or {}
+        self.cs = cs
+
+    def wavelength_rel(self) -> float:
+        """
+        Calculates the wavelength of the electron beam with relativstic kinematic effects taken
+        into account (electrons are way faster than X-rays, so you can't neglect these effects).
+        Args:
+            none
+        Returns:
+            relativisticWavelength (in meters)
+        """
+        if self.voltage in self.wavelength_cache:
+            return self.wavelength_cache[self.voltage]
+        wavelength_rel = self.planck/math.sqrt(2*self.restMassElec*self.chargeElec*1000*self.voltage*
+        	(1+(self.chargeElec*1000*self.voltage)/(2*self.restMassElec*self.speedLight**2)))
+        self.wavelength_cache[self.voltage] = wavelength_rel
+        return wavelength_rel
+
+    def generate_points(self, coord_left: int = -10, coord_right: int = 10) -> List[Tuple[int, int, int]]:
+        """
+        Generates a bunch of 3D points that span a cube.
+        Args:
+            coord_left (int): The minimum coordinate value.
+            coord_right (int): The maximum coordinate value.
+        Returns:
+            list of 3-tuples
+        """
+        points = []
+        coord_values = range(coord_left, coord_right + 1)
+        for x in coord_values:
+            for y in coord_values:
+                for z in coord_values:
+                    points.append((x,y,z))
+        return points
+            
+    def zone_axis_filter(self, points: List[Tuple[int, int, int]], laue_zone: int = 0) -> List[Tuple[int, int, int]]:
+        """
+        Filters out all points that exist within the specified Laue zone according to the zone axis rule.
+        Args:
+            points (List[Tuple[int, int, int]]): The list of points to be filtered.
+            laue_zone (int): The desired Laue zone.
+        Returns:
+            list of 3-tuples
+        """
+        # TODO: memoize this as a dict of laue zone to pointlist. prob want a system where if you input a diff range of 
+        # initial points, you edit the cache entry.
+
+        observed_points = []
+        for point in points:
+            if np.dot(self.beam_direction, np.transpose(point)) == laue_zone:
+                observed_points.append(point)
+        return observed_points
+
+    def dist_cubic(self, structure, point):
```

Comment:
The interplanar distance for any crystal system is well defined as 1/g_hkl. There is no need for specific methods for different crystal structures. See slide 19 of https://www.slideshare.net/shyuep/ucsd-nano106-03-lattice-directions-and-planes-reciprocal-lattice-and-coordinate-transformations. You simply need to compute the reciprocal lattice vector and take the norm and then the reciprocal of that, instead of relying on the lattice parameters. Alternatively, you can use the metric tensor and take the inverse to compute the reciprocal metric tensor.